Client/Frontend/Browser/TabManager.swift

```diff
@@ -436,22 +436,21 @@ class TabManager: NSObject {
         }
     }
 
-    func isSelectedParentTab(afterRemoving tab: Tab) -> Bool {
-        let viableTabs: [Tab] = tab.isPrivate ? privateTabs : normalTabs
+    // Select the most recently visited tab, IFF it is also the parent tab of the closed tab.
+    func selectParentTab(afterRemoving tab: Tab) -> Bool {
+        let viableTabs = tab.isPrivate ? privateTabs : normalTabs
+        guard let parentTab = tab.parent, parentTab != tab, !viableTabs.isEmpty else { return false }
 
-        if let parentTab = tab.parent,
-            let newTab = viableTabs.reduce(viableTabs.first, { currentBestTab, tab2 in
-                if let tab1 = currentBestTab, let time1 = tab1.lastExecutedTime {
-                    if let time2 = tab2.lastExecutedTime {
-                        return time1 <= time2 ? tab2 : tab1
-                    }
-                    return tab1
-                } else {
-                    return tab2
-                }
-            }), parentTab == newTab, tab !== newTab, newTab.lastExecutedTime != nil {
-            // We select the most recently visited tab, only if it is also the parent tab of the closed tab.
-            _selectedIndex = tabs.index(of: newTab) ?? -1
+        var parentTabIsMostRecentUsed = true
+        for candidate in viableTabs {
+            if let time = candidate.lastExecutedTime, time < (tab.lastExecutedTime ?? 0) {
+                parentTabIsMostRecentUsed = false
+                break
+            }
+        }
+
+        if parentTabIsMostRecentUsed, parentTab.lastExecutedTime != nil {
+            selectTab(parentTab, previous: tab)
```

Comment:
Direct modification of _selectedIndex is risky. Anyway, just this line is the fix, the rest of the code is a cleanup for readability. In this case, the HOF obscured the flow of the function, but that is an opinion of course. At minimum, the cleanup could be to have a guard precondition at the top and remove the if-statement complexity.